openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -110,7 +112,8 @@ def update_user_preferences(requesting_user, update, username=None):
         PreferenceUpdateError: the operation failed when performing the update.
         UserAPIInternalError: the operation failed due to an unexpected error.
     """
-    existing_user = _get_user(requesting_user, username)
+    if not existing_user:
```

Comment:
_get_user also does some authorization checks that I would prefer not be skipped.

Instead of making this API change to add another variant of user (which is very confusing and also has the issue of skipping authorization), please change the update_user_preferences method so that it requires a user object instead of a username as the third argument. The method is only used in one other (non-test) place.